### DISCUSSION TITLE
periodic-uplink example - LORAWAN_DEFAULT_DATARATE contrary to guidance

### DISCUSSION THREAD

**Discussion by dmeehan1968** [Lora-net/LoRaMac-node#1010] [Q&A]
The example sets DR_0 (Spreading Factor 12) as the default, although notes that its only used in the case of ADR being disabled.

However, this is not the case.  The join request uses the configured data rate, as do any following uplinks (even though ADR might try and adapt in the normal course).

The following document from LoRa Alliance suggests that DR0 and 1 should only be used during ADR, and not as a directly requested uplink rate.

https://lora-alliance.org/sites/default/files/2018-11/Oct122018_NetID_Alloc_Policy_Application_V3.pdf

**Comment by mluis1** [comment]
Your analysis is correct. The `LORAWAN_DEFAULT_DATARATE` applies to Join Requests and uplinks when ADR is off.

The document you refer to provides some recommendations.
Nothing prevents an end-device to use DR0 and DR1 if the environmental conditions requires it.

The `periodic-uplink-lpp` example is just an example. By default the example enables the ADR and it is up to the network server to instruct the end-device to change it data rate according to the current RF parameters environment.

In the future it would be nice if you could post this kind of questions on the project [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab. It is a better place to engage discussions and then we can agree if it is an issue or not.
If you don't mind I move this issue to the [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab.
